cclib/method/moments.py

```diff
@@ -0,0 +1,131 @@
+
+        triu_idxs = numpy.triu_indices_from(Q)
+        raveled_idxs = numpy.ravel_multi_index(triu_idxs, Q.shape)
+        quadrupole = numpy.take(Q.flatten(), raveled_idxs)
+
+        return convertor(quadrupole, 'ebohr2', 'Buckingham')
+    
+    def calculate(self, origin='nuccharge', population='mulliken',
+                  masses=None):
+        """Calculate electric dipole and quadrupole moments using parsed
+        partial atomic charges.
+        
+        Inputs:
+            origin - a choice of the origin of coordinate system. Can be
+                either a three-element iterable or a string. If
+                iterable, then it explicitly defines the origin (in
+                Angstrom). If string, then the value can be any one of
+                the following and it describes what is used as the
+                origin:
+                    * 'nuccharge' -- center of positive nuclear charge
+                    * 'mass' -- center of mass
+            population - a type of population analysis used to extract
+                corresponding atomic charges from the output file.
+            masses - if None, then use default atomic masses. Otherwise,
+                the user-provided will be used.
+
+        Returns:
+            A list where the first element is the origin of coordinates,
+            while other elements are dipole and quadrupole moments
+            expressed in terms of Debye and Buckingham units
+            respectively.
+
+        Notes:
+            To calculate the quadrupole moment the Buckingham definition
+            [1]_ is chosen. Hirschfelder et al. [2]_ define it two times
+            as much.
+
+        References:
+         .. [1] Buckingham, A. D. (1959). Molecular quadrupole moments.
+            Quarterly Reviews, Chemical Society, 13(3), 183.
+            https://doi.org:10.1039/qr9591300183.
+         .. [2] Hirschfelder J. O., Curtiss C. F. and Bird R. B. (1954).
+            The Molecular Theory of Gases and Liquids. New York: Wiley.
+        """
+        coords = self.data.atomcoords[-1]
+        try:
+            charges = self.data.atomcharges[population]
+        except KeyError:
+            raise ValueError
+
+        if hasattr(origin, '__iter__') and not isinstance(origin, str):
+            origin_pos = numpy.asarray(origin)
+        elif origin == 'nuccharge':
+            origin_pos = numpy.average(coords, weights=self.data.atomnos, axis=0)
+        elif origin == 'mass':
+            if masses:
+                atommasses = numpy.asarray(masses)
+            else:
+                try:
+                    atommasses = self.data.atommasses
+                except AttributeError:
+                    raise ValueError(
+                        "atomic masses were not parsed, consider provide "
+                        "'masses' argument instead")
+            origin_pos = numpy.average(coords, weights=atommasses, axis=0)
+        else:
+            raise ValueError
```

Comment:
Please also the same custom exception here, maybe call it `CclibMomentError`?

Reply:
Wouldn't `cclib.method.moments.CclibMomentError` name be somewhat stuttering? The `ValueError` from its [description](https://docs.python.org/3/library/exceptions.html#ValueError) doesn't fully suite our error case. Maybe `cclib.method.moments.ArgumentError`?